Still have one failing:
```
ZeroMqMessageProducerTests > testMessageProducerForPubSubDisabledWrapTopic() FAILED
    reactor.core.Exceptions$ReactorRejectedExecutionException at ZeroMqMessageProducerTests.java:188
        Caused by: java.util.concurrent.RejectedExecutionException at ZeroMqMessageProducerTests.java:188
```

--- a/spring-integration-zeromq/src/test/java/org/springframework/integration/zeromq/inbound/ZeroMqMessageProducerTests.java
+++ b/spring-integration-zeromq/src/test/java/org/springframework/integration/zeromq/inbound/ZeroMqMessageProducerTests.java
@@ -32,6 +32,7 @@ import reactor.test.StepVerifier;
 import org.springframework.beans.factory.BeanFactory;
 import org.springframework.integration.channel.FluxMessageChannel;
 import org.springframework.integration.test.util.TestUtils;
+import org.springframework.integration.zeromq.ZeroMqHeaders;
 import org.springframework.messaging.support.GenericMessage;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -158,18 +159,13 @@ public class ZeroMqMessageProducerTests {
 
 		StepVerifier stepVerifier =
 				StepVerifier.create(outputChannel)
-						.assertNext((message) ->
-								assertThat(message.getPayload())
-										.asInstanceOf(InstanceOfAssertFactories.type(ZMsg.class))
-										.extracting(ZMsg::pop)
-										.isEqualTo(new ZFrame("testTopicWithNonWrappedTopic")))
+						.assertNext((message) -> assertThat(message.getHeaders()).containsEntry(ZeroMqHeaders.TOPIC, "testTopicWithNonWrappedTopic"))
 						.thenCancel()
 						.verifyLater();
 
 		ZeroMqMessageProducer messageProducer = new ZeroMqMessageProducer(CONTEXT, SocketType.SUB);
 		messageProducer.setOutputChannel(outputChannel);
 		messageProducer.setTopics("test");
-		messageProducer.setReceiveRaw(true);
 		messageProducer.setConnectUrl(socketAddress);
 		messageProducer.setConsumeDelay(Duration.ofMillis(10));
 		messageProducer.setBeanFactory(mock(BeanFactory.class));
@@ -180,7 +176,7 @@ public class ZeroMqMessageProducerTests {
 		assertThat(socket.recv()).isNotNull();
 
 		ZMsg msg = ZMsg.newStringMsg("test");
-		msg.push(new ZFrame("testTopicWithNonWrappedTopic"));
+		msg.push("testTopicWithNonWrappedTopic");
 		msg.send(socket);
 
 		stepVerifier.verify(Duration.ofSeconds(10));

--- a/spring-integration-zeromq/src/test/java/org/springframework/integration/zeromq/inbound/ZeroMqMessageProducerTests.java
+++ b/spring-integration-zeromq/src/test/java/org/springframework/integration/zeromq/inbound/ZeroMqMessageProducerTests.java
@@ -90,7 +90,7 @@ public class ZeroMqMessageProducerTests {
 
 		stepVerifier.verify();
 
-		messageProducer.destroy();
+		messageProducer.stop();
 		socket.close();
 	}
 
@@ -144,7 +144,7 @@ public class ZeroMqMessageProducerTests {
 
 		stepVerifier.verify(Duration.ofSeconds(10));
 
-		messageProducer.destroy();
+		messageProducer.stop();
 		socket.close();
 	}
 
@@ -153,7 +153,6 @@ public class ZeroMqMessageProducerTests {
 		String socketAddress = "inproc://messageProducerWrapTopic.test";
 		ZMQ.Socket socket = CONTEXT.createSocket(SocketType.XPUB);
 		socket.bind(socketAddress);
-		socket.setReceiveTimeOut(10_000);
 
 		FluxMessageChannel outputChannel = new FluxMessageChannel();
 
@@ -167,7 +166,6 @@ public class ZeroMqMessageProducerTests {
 		messageProducer.setOutputChannel(outputChannel);
 		messageProducer.setTopics("test");
 		messageProducer.setConnectUrl(socketAddress);
-		messageProducer.setConsumeDelay(Duration.ofMillis(10));
 		messageProducer.setBeanFactory(mock(BeanFactory.class));
 		messageProducer.unwrapTopic(false);
 		messageProducer.afterPropertiesSet();
@@ -179,9 +177,9 @@ public class ZeroMqMessageProducerTests {
 		msg.push("testTopicWithNonWrappedTopic");
 		msg.send(socket);
 
-		stepVerifier.verify(Duration.ofSeconds(10));
+		stepVerifier.verify();
 
-		messageProducer.destroy();
+		messageProducer.stop();
 		socket.close();
 	}
 

